01-reading.Rmd

```diff
@@ -614,23 +711,148 @@ Next, we tell R what page we want to scrape by providing the webpage's URL in qu
+grants you API access, they also usually specify a limit (or *quota*) of how much data you can ask for.
+Be careful not to overrun your quota! In this example, we should take a look at
+ [the Twitter website](https://developer.twitter.com/en/docs/twitter-api/rate-limits) to see what limits
+we should abide by when using the API. 
+
+After checking the website, it seems like asking for 400 tweets one time is acceptable.
+So we can use the `get_timelines` function to ask for the last 400 tweets from the Tidyverse account.
+```r
+rt <- get_timelines('tidyverse', n=400)
+```
+
+When you call the `get_timelines` for the first time (or any other `rtweet` function that accesses the API), 
+you will see a browser pop-up that looks something like this:
+```{r 01-tidyverse-authorize, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "The rtweet authorization prompt.", fig.retina = 2, out.width="800"}
+knitr::include_graphics("img/authorize_question.png")
+```
+
+This is the `rtweet` package asking you to provide your own Twitter account's login information.
+When `rtweet` talks to the Twitter API, it uses your account information to authenticate requests;
+Twitter then can keep track of how much data you're asking for, and how frequently you're asking.
+If you want to follow along with this example using your own Twitter account, you should read
+over the list of permissions you are granting `rtweet` *very carefully* and make sure you are comfortable
+with it. Note that `rtweet` can be used to manage most aspects of your account (make posts, add friends, etc.),
+which is why `rtweet` asks for such extensive permissions.
+If you decide to allow `rtweet` to talk to the Twitter API using your account information, then 
+input your username and password and hit "Sign In." Twitter will probably send you an email to say
+that there was an unusual login attempt on your account, and in that case you will have to take
+the one-time code they send you and provide that to the `rtweet` login page too. 
+
+> Every API has its own way to authenticate users when they try to access data. Many APIs require you to
+> sign up to receive a *token*, which is a secret password that you input into the R package (like `rtweet`) 
+> that you are using to access the API. 
+
+With the authentication setup out of the way, let's run the `get_timelines` function again to actually access
+the API and take a look at what was returned:
+```r
+rt <- get_timelines('tidyverse', n=400)
+rt
+```
+```{r 01-reading-hidden-load-tweets, echo = FALSE, message = FALSE, warning = FALSE}
+rt <- read_csv("data/tweets.csv")
+rt
+```
+The data has quite a few variables! Let's reduce this down to a few variables of interest: `created_at`, `text`, `is_retweet`, and `retweet_screen_name`.
+```{r 01-select-tweets, message = FALSE, warning = FALSE}
+rt <- rt %>% select(created_at, text, is_retweet, retweet_screen_name)
+rt
+```
+If you look back up at the image of the Tidyverse twitter page, you will
```

Comment:
I think the format of data retrieved from this API is in stark contrast to the format of data retrieved from the web scraping example. We should make a comment here about this. In particular, the readers might think that API's always give back data frames. This is not true. So a comment stating that API's generally give you back more homogenous (or organized?) data with less code as compared to web scraping, is fair, I think. But I think we're also warranted to state that different data formats will be given by different API's, or even different functions within the same API!

Reply:
resolved via commit